src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/CompareConverter.shared.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions.Internals;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	/// Converts an object that implements IComparable to an object or a boolean based on a comparaison.
+	/// </summary>
+	public class CompareConverter : CompareConverter<object>
+	{
+	}
+
+	/// <summary>
+	/// Converts an object that implements IComparable to an object or a boolean based on a comparaison.
+	/// </summary>
+	public class CompareConverter<TObject> : ValueConverterExtension, IValueConverter
+	{
+		[Flags]
+		public enum Operator
+		{
+			Greater,
+			GreaterOrEqual,
+			Equal,
+			NotEqual,
+			SmallerOrEqual,
+			Smaller,
+		}
+
+		enum Modes
+		{
+			Boolean,
+			Object
+		}
+
+		Modes mode;
+
+		/// <summary>
+		/// The comparing value.
+		/// </summary>
+		public IComparable? ComparingValue { get; set; }
+
+		/// <summary>
+		/// The comparison operator.
+		/// </summary>
+		public Operator ComparisonOperator { get; set; }
+
+		/// <summary>
+		/// The object that corresponds to True value.
+		/// </summary>
+		public TObject? TrueObject { get; set; }
+
+		/// <summary>
+		/// The object that corresponds to False value.
+		/// </summary>
+		public TObject? FalseObject { get; set; }
+
+		/// <summary>
+		/// Converts an object that implements IComparable to a specified object or a boolean based on a comparaison result.
+		/// </summary>
+		/// <param name="value">The value to convert.</param>
+		/// <param name="targetType">The type of the binding target property. This is not implemented.</param>
+		/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
+		/// <param name="culture">The culture to use in the converter.  This is not implemented.</param>
+		/// <returns>The object assigned to <see cref="TrueObject"/> if (value <see cref="ComparisonOperator"/> <see cref="ComparingValue"/>) equals True and <see cref="TrueObject"/> is not null, if <see cref="TrueObject"/> is null it returns true, otherwise the value assigned to <see cref="FalseObject"/>, if no value is assigned then it returns false.</returns>
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
```

Comment:
Since we allow inheritance from this class, let's make this `virtual`
```suggestion
		public virtual object Convert(object value, Type targetType, object parameter, CultureInfo culture)
```

Reply:
Could you provide a use case where this needs to be overridden?

Reply:
you do it at line 11, so the dev can take that as inspiration and want to use some strongly typed type, and for it the user maybe can override those methods to add his on implementation